src/error.rs

```diff
@@ -28,6 +26,10 @@ pub enum Error {
     DatabaseError(#[from] DatabaseError),
     #[error("Provider error: {0}")]
     ProviderError(#[from] ProviderError),
+    #[error("Reference source returned a key corresponding to a null record: {0}")]
+    UnexpectedNullRemoteFromProvider(String),
+    #[error("Reference source returned a remote id which is invalid: {0}")]
+    InvalidRemoteIdFromProvider(RecordError),
```

Comment:
Should these error variants be merged into the `ProviderError`?

Reply:
I think they should.